tcwebhooks-web-ui/src/main/resources/buildServerResources/WebHook/js/editWebhook.js

```diff
@@ -536,22 +536,25 @@ function populateWebHookDialog(id){
 			$j('#webHookFormContents select#payloadFormatHolder').val(webhook.payloadTemplate + "_" + webhook.payloadFormat).change();
 			
 			$j('#buildTypeSubProjects').prop('checked', webhook.subProjectsEnabled);
-			$j.each(webhook.builds, function(){
-				var thing = $j(this.buildTypeName).text();
-				console.log(thing);
-				 if (this.enabled){
-			 	 	$j('#buildList').append('<p style="border-bottom:solid 1px #cccccc; margin:0; padding:0.5em;"><label><input checked onclick="updateSelectedBuildTypes();" type=checkbox style="padding-right: 1em;" name="buildTypeId" value="' + this.buildTypeId + '"class="buildType_single">' + htmlEscape(this.buildTypeName) + '</label></p>');
-				 } else {
-				 	 $j('#buildList').append('<p style="border-bottom:solid 1px #cccccc; margin:0; padding:0.5em;"><label><input onclick="updateSelectedBuildTypes();" type=checkbox style="padding-right: 1em;" name="buildTypeId" value="' + this.buildTypeId + '"class="buildType_single">' + htmlEscape(this.buildTypeName) + '</label></p>');
-				 }
+			$j.each(webhook.builds, function() {
+				console.log('Adding subproject', this.buildTypeId, 'named', this.buildTypeName);
+				var isChecked = '';
+				if (this.enabled) {
+					isChecked = ' checked';
+				}
+				var label = $j('<label><input' + isChecked + ' onclick="updateSelectedBuildTypes();" type=checkbox style="padding-right: 1em;" name="buildTypeId" value="' + this.buildTypeId + '"class="buildType_single"></label>');
+				label.text(this.buildTypeName);
```

Comment:
Perhaps this line clears the label content (child item) and replaces it with the text.

Reply:
Yes.. yes that is exactly what it is doing. *facepalm\* I can't believe I missed that. OK, I'll fix it.